firecrown/tests/test_integration_generic_ccl.py

```diff
@@ -0,0 +1,266 @@
+    for i, mn in enumerate([0.25, 0.75]):
+        z = np.linspace(0, 2, 50)
+        dndz = np.exp(-0.5 * (z - mn)**2 / 0.25 / 0.25)
+
+        sacc_data.add_tracer('NZ', 'trc%d' % i, z, dndz)
+
+        tracers.append(ccl.WeakLensingTracer(
+            cosmo,
+            dndz=(z, dndz)))
+
+    dv = []
+    ndv = []
+    for i in range(len(tracers)):
+        for j in range(i, len(tracers)):
+            ell = np.logspace(1, 4, 10)
+            pell = ccl.angular_cl(cosmo, tracers[i], tracers[j], ell)
+            npell = pell + rng.normal(size=pell.shape[0]) * eps * pell
+
+            sacc_data.add_ell_cl(
+                'galaxy_shear_cl_ee', 'trc%d' % i, 'trc%d' % j, ell, npell)
+            dv.append(pell)
+            ndv.append(npell)
+
+    mn = 0.25
+    z = np.linspace(0, 2, 50)
+    dndz = np.exp(-0.5 * (z - mn)**2 / 0.25 / 0.25)
+    dndz /= np.max(dndz)
+    sacc_data.add_tracer(
+        'NZ', 'cl1', z, dndz,
+        metadata={"lnlam_min": np.log(1e14), "lnlam_max": np.log(1e16), 'area_sd': 200})
+
+    intp = Akima1DInterpolator(z, dndz)
+
+    def _sel(m, a):
+        a = np.atleast_1d(a)
+        m = np.atleast_1d(m)
+        z = 1.0 / a - 1.0
+        logm = np.log10(m)
+        zsel = intp(z)
+        msk = ~np.isfinite(zsel)
+        zsel[msk] = 0.0
+        vals = np.zeros((m.shape[0], a.shape[0]))
+        vals[:] = zsel
+        msk = (logm >= 14) & (logm < 16)
+        vals[~msk, :] = 0
+        return vals
+
+    mdef = ccl.halos.MassDef(200, 'matter')
+    hmf = ccl.halos.MassFuncTinker10(cosmo, mdef,
+                                     mass_def_strict=False)
+    hbf = ccl.halos.HaloBiasTinker10(cosmo, mass_def=mdef,
+                                     mass_def_strict=False)
+
+    hmc = ccl.halos.HMCalculator(
+        cosmo, hmf, hbf, mdef,
+        integration_method_M='spline',
+        nlog10M=256
+    )
+
+    true_cnts = hmc.number_counts(cosmo, _sel, amin=0.333333, amax=1, na=256)
+    true_cnts *= (200 * (np.pi/180.0)**2)
+    ntrue_cnts = true_cnts + rng.normal() * eps * true_cnts
+    sacc_data.add_data_point(
+        "count",
+        ("cl1",),
+        ntrue_cnts,
+    )
+
+    dv.append(np.array([true_cnts]))
+    ndv.append(np.array([ntrue_cnts]))
+
+    # a fake covariance matrix
+    dv = np.concatenate(dv, axis=0)
+    ndv = np.concatenate(ndv, axis=0)
+    cov = np.zeros((dv.shape[0], dv.shape[0]))
+    for i in range(len(dv)):
+        cov[i, i] = (eps * dv[i]) ** 2
+    sacc_data.add_covariance(cov)
+
+    sacc_data.save_fits(os.path.join(tmpdir, 'sacc.fits'), overwrite=True)
+
+    cinv = np.linalg.inv(cov)
+    delta = ndv - dv
+    loglike = -0.5 * np.dot(delta, np.dot(cinv, delta))
+
+    config = """\
+parameters:
+  Omega_k: 0.0
+  Omega_c: 0.27
+  Omega_b: 0.045
+  h: 0.67
+  n_s: [0.9, 0.96, 1.0]
+  sigma8: 0.8
+  w0: -1.0
+  wa: 0.0
+  transfer_function: 'eisenstein_hu'
+
+  # lens bin zero
+  src0_delta_z: [-0.1, 0.0, 0.1]
+  src1_delta_z: 0.0
+
+two_point_plus_clusters:
```

Comment:
Is this lensing around the clusters?

Reply:
No. This is just a random name in the file.